### PR TITLE
feat: default 8gb hard memory limit for redis

### PR DESCRIPTION
**Pull request by sunnygleason**
#### Problem

* redis overflows on testnets because a default limit isn't set 😞 

#### Summary of Changes

* use sed to update the limit ✨ 

Fixes # N/A

### CODE REVIEW

**Comment by sunnygleason** [comment]
@mvines I think this should do the trick...

**Comment by sunnygleason** [comment]
@mvines I'll wait until you're around to merge & we can observe impact on test/beta/etc...

File: net/scripts/install-redis.sh [hunk 1]
```diff
@@ -8,3 +8,7 @@ add-apt-repository -y ppa:chris-lea/redis-server
 apt-get --assume-yes install redis
 
 systemctl enable redis-server.service
+
+sed -i '/^maxmemory .*/ s//maxmemory 8gb # limit set by solana\/net\/scripts\/install-redis.sh/' /etc/redis/redis.conf
```
Comment (mvines) [inline net/scripts/install-redis.sh]: Would redis barf if we did this the braindead way?
```bash
cat "maxmemory 8gb" >> /etc/redis/redis.conf
```
I'm worried about the case of a redis.conf that doesn't contain `/^maxmemory .*/`

Comment (sunnygleason) [reply net/scripts/install-redis.sh]: maybe do a bash `if grep` block here? I don't know it offhand, but if you do, feel free to share -- if not, I can look it up!

Comment (sunnygleason) [reply net/scripts/install-redis.sh]: (I'll do the same for memory eviction -- it's also set in the default redis config but if we're assuming an unfriendly config, I'll take care of it for both)

Comment (mvines) [reply net/scripts/install-redis.sh]: Not sure off hand.  An `if grep` would work too but is more code than the braindead way (if it works!)

Comment (sunnygleason) [reply net/scripts/install-redis.sh]: ok I'll try to wrangle up something like this:

image

File: net/scripts/install-redis.sh [hunk 2]
```diff
@@ -8,3 +8,38 @@ add-apt-repository -y ppa:chris-lea/redis-server
...
 
 systemctl enable redis-server.service
+
+
+
+REDIS_CONF=/etc/redis/redis.conf
+
+FOUND1=$(grep "^maxmemory " $REDIS_CONF)
+
+if [ -n "$FOUND1" ]; then
```
Comment (mvines) [inline net/scripts/install-redis.sh]:
```bash
if grep -q "^maxmemory " $REDIS_CONF; then
```

File: net/scripts/install-redis.sh [hunk 3]
```diff
@@ -8,3 +8,38 @@ add-apt-repository -y ppa:chris-lea/redis-server
 apt-get --assume-yes install redis
 
 systemctl enable redis-server.service
+
+
```
Comment (mvines) [inline net/scripts/install-redis.sh]: nit: we don't need this much whitespace

File: net/scripts/install-redis.sh [hunk 4]
```diff
@@ -8,3 +8,38 @@ add-apt-repository -y ppa:chris-lea/redis-server
...
+REDIS_CONF=/etc/redis/redis.conf
+
+FOUND1=$(grep "^maxmemory " $REDIS_CONF)
+
+if [ -n "$FOUND1" ]; then
+  echo "setting maxmemory"
+  sed -i '/^maxmemory .*/ s//maxmemory 8gb/' $REDIS_CONF
+else
+  echo "maxmemory not present: appending setting"
+cat << EOF >> $REDIS_CONF
```
Comment (mvines) [inline net/scripts/install-redis.sh]: nit: indent `cat`

File: net/scripts/install-redis.sh [hunk 5]
```diff
@@ -8,3 +8,38 @@ add-apt-repository -y ppa:chris-lea/redis-server
...
+fi
+
+FOUND2=$(grep "^maxmemory-policy " $REDIS_CONF)
+
+if [ -n "$FOUND2" ]; then
+  echo "setting maxmemory-policy"
+  sed -i '/^maxmemory-policy .*/ s//maxmemory-policy allkeys-lru/' $REDIS_CONF
+else
+  echo "maxmemory-policy not present: appending setting"
+cat << EOF >> $REDIS_CONF
```
Comment (mvines) [inline net/scripts/install-redis.sh]: nit: indent `cat`

**Comment by sunnygleason** [comment]
@mvines merging FYI, lemme know if you need any more work in this area!

**Comment by sunnygleason** [comment]
(🕐 ... waiting for backport CI)  😄

**Pull request merged**